testsuite/regressiontests/unused_names.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+#
+# unused_names.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import sys
+
+# Use encoding-aware Py3 open also in Py2
+if sys.version_info[0] < 3:
+    from io import open
+
+EXIT_SUCCESS = 0
+EXIT_NAME_H_CPP_MISMATCH = 30
+EXIT_UNUSED_NAME = 31
+EXIT_NO_SOURCE = 126
+
+try:
+    source_dir = os.environ['NEST_SOURCE']
+except KeyError:
+    print("Please make NEST_SOURCE environment variable to point to " +
+          "the source tree you want to check!")
+    sys.exit(EXIT_NO_SOURCE)
+
+# Base names of files that contain const Name definitions
+# (i.e. filenames without filename extension)
+names_files = [
+    "librandom/librandom_names",
+    "nestkernel/nest_names",
+    "topology/topology_names",
+]
+
+names_defined = set()
```

Comment:
I would make this a list. Then you can test later if `names_defined == unique(names_defined)`. This catches cases where automated merges duplicate lines---has happened a few times.

Reply:
I've refactored this together with the check for unused names in 5a58f65026e and think that catching errors with automatic merges (and other unrelated things) should not be covered by this test if it does not come for free.

Reply:
Ok.